cypher-manual/clauses/match.md

```diff
@@ -320,6 +324,112 @@ Output:
 +----------------------------------------+
 ```
 
+## 4. Using multiple `MATCH` clauses
+
+### 4.1. Cartesian product of nodes
+
+To match each person from your dataset with each of the countries in Europe, you can use multiple `MATCH` clauses as following: 
+
+```cypher
+MATCH (p:Person)
+MATCH (c:Country {continent: "Europe"})
+RETURN p.name,c.name;
+```
+
+Output: 
+```nocopy 
++------------------+------------------+
+| p.name           | c.name           |
++------------------+------------------+
+| "John"           | "Germany"        |
+| "Harry"          | "Germany"        |
+| "Anna"           | "Germany"        |
+| "John"           | "France"         |
+| "Harry"          | "France"         |
+| "Anna"           | "France"         |
+| "John"           | "United Kingdom" |
+| "Harry"          | "United Kingdom" |
+| "Anna"           | "United Kingdom" |
++------------------+------------------+
+```
+
+The query returns cartesian product of matched nodes. The output of the first `MATCH` clause is matched with each output of the second `MATCH` clause. In our case, each person from our dataset is matched with each country in *Europe*. 
+
+### 4.2. Creating a list
+
+If you want to create a list containing the results of different `MATCH` queries, you can achieve that with multiple `MATCH` clauses in one query:
+
+```cypher
+MATCH (p:Person)
+WITH COLLECT(p.name) as people
+MATCH (c:Country {continent: "Europe"})
+WITH people + COLLECT(c.name) as names
+RETURN names;
+```
+
+Output:
+```nocopy
++------------------------------------------------------------------+
+| names                                                            |
++------------------------------------------------------------------+
+| ["John", "Harry", "Anna", "Germany", "France", "United Kingdom"] |
++------------------------------------------------------------------+
+```
+
+The query returns a list of names of all people from our dataset concatenated with the names of all of the countries in Europe. 
+
+:::caution
+
+If any of the sets `MATCH` clause returns is empty, the whole output will be an empty list.
+
+:::
+
+If you run the similar query again:
+
+```cypher
+MATCH (p:Person)
+WITH COLLECT(p.name) as people
+MATCH (c:Country {continent: "Asia"})
+WITH people + COLLECT(c.name) as names
+RETURN names;
+```
+
+Output:
+
+```nocopy
++-------+
+| names |
++-------+
+| Null  |
++-------+
+```
+
+Since we don't have any node labeled Country with a property continent named *Asia*, the second `MATCH` clause returns an empty dataset and therefore the output is also an empty list. To avoid that, use `OPTIONAL MATCH` clause.
+
+### 4.3. Using the `OPTIONAL MATCH` clause
+
+If we run the same query as before, only using the `OPTIONAL MATCH` clause: 
```

Comment:
```suggestion
To avoid getting an empty list as an output, due to any of the `MATCH` clauses returning an empty set,  use the `OPTIONAL MATCH` clause: 
```